platform/config/ensureDeepObject.ts

```diff
@@ -1,29 +1,42 @@
 const separator = '.';
 
+/**
+ * Recursively traverses through the object's properties and expands ones with
+ * dot-separated names into nested objects (eg. { a.b: 'c'} -> { a: { b: 'c' }).
+ * @param {*} obj Object to traverse through
+ * @returns {Object} Same object instance with expanded properties.
```

Comment:
I think we can remove the types from jsdoc for now, then look deeper into it later. Should be easy to add back based on the TS types.